src/qt/treectrl.cpp

```diff
@@ -56,275 +495,699 @@ void wxTreeCtrl::SetIndent(unsigned int indent)
+    qTreeItem->setBackgroundColor(0, wxQtConvertColour(col));
 }
 
 void wxTreeCtrl::SetItemFont(const wxTreeItemId& item, const wxFont& font)
 {
+    wxCHECK_RET(item.IsOk(), "invalid tree item");
+
+    QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    qTreeItem->setFont(0, font.GetHandle());
 }
 
 bool wxTreeCtrl::IsVisible(const wxTreeItemId& item) const
 {
-    return false;
+    wxCHECK_MSG(item.IsOk(), false, "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    const QRect visualRect = m_qtTreeWidget->visualItemRect(qTreeItem);
+    return visualRect.isValid();
 }
 
 bool wxTreeCtrl::ItemHasChildren(const wxTreeItemId& item) const
 {
-    return false;
+    wxCHECK_MSG(item.IsOk(), false, "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    return qTreeItem->childCount() > 0;
 }
 
 bool wxTreeCtrl::IsExpanded(const wxTreeItemId& item) const
 {
-    return false;
+    wxCHECK_MSG(item.IsOk(), false, "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    return qTreeItem->isExpanded();
 }
 
 bool wxTreeCtrl::IsSelected(const wxTreeItemId& item) const
 {
-    return false;
+    wxCHECK_MSG(item.IsOk(), false, "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    return qTreeItem->isSelected();
 }
 
 bool wxTreeCtrl::IsBold(const wxTreeItemId& item) const
 {
-    return false;
+    wxCHECK_MSG(item.IsOk(), false, "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    const QFont font = qTreeItem->font(0);
+    return font.bold();
 }
 
-size_t wxTreeCtrl::GetChildrenCount(const wxTreeItemId& item, bool recursively) const
+size_t wxTreeCtrl::GetChildrenCount(
+    const wxTreeItemId& item,
+    bool recursively
+) const
 {
-    return 0;
+    wxCHECK_MSG(item.IsOk(), 0, "invalid tree item");
+
+    QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+
+    if ( recursively )
+        return CountChildren(qTreeItem);
+
+    return qTreeItem->childCount();
 }
 
 wxTreeItemId wxTreeCtrl::GetRootItem() const
 {
-    return wxTreeItemId();
+    const QTreeWidgetItem *root = m_qtTreeWidget->invisibleRootItem();
+    return wxQtConvertTreeItem(root->child(0));
 }
 
 wxTreeItemId wxTreeCtrl::GetSelection() const
 {
-    return wxTreeItemId();
+    QList<QTreeWidgetItem*> selections = m_qtTreeWidget->selectedItems();
+    return selections.isEmpty()
+        ? wxTreeItemId()
+        : wxQtConvertTreeItem(selections[0]);
 }
 
 size_t wxTreeCtrl::GetSelections(wxArrayTreeItemIds& selections) const
 {
-    return 0;
+    QList<QTreeWidgetItem*> qtSelections = m_qtTreeWidget->selectedItems();
+
+    const size_t numberOfSelections = qtSelections.size();
+    for ( size_t i = 0; i < numberOfSelections; ++i )
+    {
+        QTreeWidgetItem *item = qtSelections[i];
+        selections.Add(wxQtConvertTreeItem(item));
```

Comment:
Please prefer std-like methods in the new code:
```suggestion
        selections.push_back(wxQtConvertTreeItem(item));
```